.github/workflows/Intel.yml

```diff
@@ -27,60 +21,29 @@ jobs:
 
     steps:
 
-    - uses: actions/checkout@v2
+    - name: checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: true
 
       # See https://software.intel.com/content/www/us/en/develop/articles/oneapi-repo-instructions.html
-    - name: install-intel
-      run: |
-        cd /tmp
-        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
-        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
-        rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
-        echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
-        sudo apt-get update
-        sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-openmp intel-oneapi-mpi intel-oneapi-mpi-devel intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic intel-oneapi-compiler-fortran
-        echo "source /opt/intel/oneapi/setvars.sh" >> ~/.bash_profile
-        if [ ${{ matrix.compilers }} == "oneapi" ]; then
-          echo "export CC=mpiicx FC=mpiifx" >> ~/.bash_profile
-        elif [ ${{ matrix.compilers }} == "intel" ]; then
-          echo "export CC=mpiicc FC=mpiifort" >> ~/.bash_profile
-        fi
-
-    - name: checkout-bacio
-      uses: actions/checkout@v2
+    - name: "Install Intel"
+      uses: NOAA-EMC/ci-install-intel-toolkit@develop
       with:
-        repository: NOAA-EMC/NCEPLIBS-bacio
-        path: bacio
-        ref: develop
+        install-mpi: true
+        mpi-wrapper-setup: ${{ matrix.compilers }}
 
-    - name: build-bacio
-      run: |
-        cd bacio
-        mkdir build
-        cd build
-        cmake .. -DCMAKE_INSTALL_PREFIX=~/bacio
-        make -j2
-        make install
-
-    - name: checkout-w3emc
-      uses: actions/checkout@v2
+    - name: "Build dependencies"
+      uses: NOAA-EMC/ci-build-nceplibs@develop
       with:
-        repository: NOAA-EMC/NCEPLIBS-w3emc
-        path: w3emc
-        ref: develop
-
-    - name: build-w3emc
-      run: |
-        cd w3emc
-        mkdir build
-        cd build
-        cmake .. -DCMAKE_INSTALL_PREFIX=~/w3emc -DCMAKE_PREFIX_PATH="~/" -DBUILD_WITH_BUFR=OFF
-        make -j2
-        make install
+        bacio-version: develop
+        w3emc-version: develop
```

Comment:
Is it wise to check with another repo's `develop`?  Should this be a tag instead?
I ask because, a change in upstream may not be directly usable in the downstream.

Reply:
This is what this workflow currently does. I think there's merit both ways in terms of develop vs. using a specific release. @Hang-Lei-NOAA, any thoughts on what makes the most sense here?

Reply:
Indeed we need to test against released versions and the develop branch.